### PR TITLE
Add markdown table

### PR DESCRIPTION
**Pull request by ShalokShalom**
I hope this helps. 

Also see: https://github.com/lxc/incus/issues/186

Thanks a lot for the review
[Preview](https://github.com/lxc/incus/blob/4f056c3cd6bcda2f28b585db74012a8c6bb204ed/doc/explanation/containers_and_vms.md)

### CODE REVIEW

**Comment by ShalokShalom** [comment]
Does anybody know, what that failure is caused by: 
https://github.com/lxc/incus/actions/runs/6602346346/job/17940899712?pr=187#step:12:18967

**Comment by monstermunchkin** [comment]
> Does anybody know, what that failure is caused by: https://github.com/lxc/incus/actions/runs/6602346346/job/17940899712?pr=187#step:12:18967

That's just a random test failure. It's unrelated to your PR.

**Comment by ShalokShalom** [comment]
> > Does anybody know, what that failure is caused by: https://github.com/lxc/incus/actions/runs/6602346346/job/17940899712?pr=187#step:12:18967
> 
> That's just a random test failure. It's unrelated to your PR.

Oh thanks. Does the PR look good to you?

**Comment by stgraber** [comment]
```
.tmp/doc/explanation/containers_and_vms.md:7: MD009 Trailing spaces
.tmp/doc/explanation/containers_and_vms.md:10: MD009 Trailing spaces
.tmp/doc/explanation/containers_and_vms.md:12: MD009 Trailing spaces
.tmp/doc/explanation/containers_and_vms.md:13: MD009 Trailing spaces
.tmp/doc/explanation/containers_and_vms.md:15: MD009 Trailing spaces
.tmp/doc/explanation/containers_and_vms.md:16: MD009 Trailing spaces
.tmp/doc/explanation/containers_and_vms.md:17: MD009 Trailing spaces
.tmp/doc/explanation/containers_and_vms.md:18: MD009 Trailing spaces
.tmp/doc/explanation/containers_and_vms.md:19: MD009 Trailing spaces
.tmp/doc/explanation/containers_and_vms.md:25: MD009 Trailing spaces
.tmp/doc/explanation/containers_and_vms.md:36: MD009 Trailing spaces
```

File: doc/explanation/containers_and_vms.md [hunk 1]
```diff
@@ -3,22 +3,37 @@
 
 Incus provides support for two different types of {ref}`instances <expl-instances>`: *system containers* and *virtual machines*.
 
-When running a system container, Incus simulates a virtual version of a full operating system. To do this, it uses the functionality provided by the kernel running on the host system.
+When running a system container, Incus simulates a virtual version of a full operating system.  
+To do this, it uses the functionality provided by the kernel running on the host system.
```
Comment (stgraber) [inline doc/explanation/containers_and_vms.md:7]: Line break isn't needed as it's not visible in the rendered document and so just makes us needlessly diverge from the matching text in the LXD doc.

File: doc/explanation/containers_and_vms.md [hunk 2]
```diff
@@ -3,22 +3,37 @@
...
 
-When running a virtual machine, Incus uses the hardware of the host system, but the kernel is provided by the virtual machine. Therefore, virtual machines can be used to run, for example, a different operating system.
+When running a virtual machine, Incus uses the hardware of the host system, and the kernel is provided by the virtual machine.  
+Therefore, virtual machines can be used to run, for example, a different operating system.
```
Comment (stgraber) [inline doc/explanation/containers_and_vms.md:10]: The original phrasing with `, but the kernel is provided` sounds better to me.
Same comment as above about adding a line break, doesn't make a difference to the rendered document but makes it harder for us to keep up with changes in LXD.

File: doc/explanation/containers_and_vms.md [hunk 3]
```diff
@@ -3,22 +3,37 @@
...
+ 
+|    Virtual Machines              |  Application Containers          |     System Containers    |      
+|      :----:                     |      :----:                     |        :----:           |
+| Uses a dedicated kernel                 |  Uses the kernel of the host            |   Uses the kernel of the host   |    
+| Can host different types of OS  |  Can only host Linux            |  Can only host Linux    |   
+| Uses more ressources            |  Uses less ressources           |  Uses less resources    |   
+| Can host multiple applications         |  Can host a single app |  Can host multiple applications |
```
Comment (stgraber) [inline doc/explanation/containers_and_vms.md:17]: Probably should be `Can host a single application` for consistency with the other fields

File: doc/explanation/containers_and_vms.md [hunk 4]
```diff
@@ -3,22 +3,37 @@
...
+When running a virtual machine, Incus uses the hardware of the host system, and the kernel is provided by the virtual machine.  
+Therefore, virtual machines can be used to run, for example, a different operating system.
+ 
+|    Virtual Machines              |  Application Containers          |     System Containers    |      
+|      :----:                     |      :----:                     |        :----:           |
+| Uses a dedicated kernel                 |  Uses the kernel of the host            |   Uses the kernel of the host   |    
+| Can host different types of OS  |  Can only host Linux            |  Can only host Linux    |   
+| Uses more ressources            |  Uses less ressources           |  Uses less resources    |   
+| Can host multiple applications         |  Can host a single app |  Can host multiple applications |  
+| Supported by Incus              |  Supported by Docker            |  Supported by Incus     |
```
Comment (stgraber) [inline doc/explanation/containers_and_vms.md:18]: Maybe go with `Supported by Docker & similar` given there are other application container runtimes/managers these days.

File: doc/explanation/containers_and_vms.md [hunk 5]
```diff
@@ -3,22 +3,37 @@
...
+| Can host different types of OS  |  Can only host Linux            |  Can only host Linux    |   
+| Uses more ressources            |  Uses less ressources           |  Uses less resources    |   
+| Can host multiple applications         |  Can host a single app |  Can host multiple applications |  
+| Supported by Incus              |  Supported by Docker            |  Supported by Incus     |  
 
-## Application containers vs. system containers
+## Application containers and system containers
 
-Application containers (as provided by, for example, Docker) package a single process or application. System containers, on the other hand, simulate a full operating system and let you run multiple processes at the same time.
+Application containers like Docker package a single process or application.
```
Comment (stgraber) [inline doc/explanation/containers_and_vms.md:22]: I preferred the previous wording as it's technically more correct.
Docker isn't an application container, it's a tool that can create and manage (provide) application containers.

**Review by stgraber — CHANGES_REQUESTED** [review]

**Comment by stgraber** [comment]
Oh, why was this one closed?

**Pull request closed without merging**